kore/test/Test/Kore/Builtin/Map.hs

```diff
@@ -1580,3 +1585,15 @@ distinctVariables variables =
     length variableNames == length (List.nub variableNames)
   where
     variableNames = map asVariableName variables
+
+-- | Utility function for mapping over the keys of a 'HashMap'.
+mapKeys ::
+    Hashable k2 =>
+    Eq k2 =>
+    (k1 -> k2) ->
+    HashMap k1 v ->
+    HashMap k2 v
+mapKeys f =
+    HashMap.fromList
```

Comment:
Can you confirm that no use of this function supplies an `f` that is non-injective? Because if it did, we would risk losing entries from the hash map.

Reply:
We only ever use this function in places where we used to use `Data.Map.Strict.mapKeys`, which has the same behavior you described.